packages/block-library/src/query-loop/index.php

```diff
@@ -26,8 +26,18 @@ function render_block_core_query_loop( $attributes, $content, $block ) {
 		'post__not_in' => array(),
 	);
 
+	$use_global_query = ( isset( $block->context['query']['customQuery'] ) && false === $block->context['query']['customQuery'] );
```

Comment:
I think the property could be renamed to `useGlobalQuery` and therefore invert the logic/checks in code. Also this should be a top level property of `Query` block (changes in block.json etc..)